Replace issue templates with new issue forms - v2

NicolasHug opened the pull request:
This PR introduces the new issue forms that replace issue templates.
(This is exactly the same as https://github.com/pytorch/pytorch/pull/65917 which was reverted due to an issue during the import)

This is similar to what was done in torchvision https://github.com/pytorch/vision/pull/4299 and torchaudio, you can see the end result here: https://github.com/pytorch/vision/issues/new/choose (click e.g. on the [bug report](https://github.com/pytorch/vision/issues/new?assignees=&labels=&template=bug-report.yml))

The main new thing is that we can enforce some of the fields to be filled, especially for bug reports. It's also a much cleaner GUI for users IMHO, and we can provide better examples and instructions.

There is still a "blank" template available.

I removed the "Questions" form: we say we close these issues anyway. I replaced it with a direct link to https://discuss.pytorch.org. Since we still have a "blank" template, I think this  covers all previous use-cases properly.


Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #69361

Differential Revision: [D32947189](https://our.internmc.facebook.com/intern/diff/D32947189)

NicolasHug committed:
Add new issue forms

[ghstack-poisoned]

NicolasHug commented on .github/ISSUE_TEMPLATE/bug-report.yml:
````diff
@@ -0,0 +1,56 @@
+      python collect_env.py
+      ```
+  validations:
+    required: true
````
The `versions` field of the bug-report is mandatory.

Versions are almost always needed in bug report, so it saves time. It also discourages low-effort or low-quality reports, which IMHO is a good thing. Feel free to remove though.

Note that it's still possible to create blank issues without a template, effectively bypassing this requirement if you want to (I personally always use the blank template).

vadimkantorov commented:
For torchvision I also found the separation between feature description and motivation/pitch a bit too duplicative for the many simple feature requests that consist only of couple sentences...

NicolasHug commented:
In the current issue template, motivation and pitch are separate (I merged them here). But I agree it's still redundant. I kept it for legacy reasons but I'd be happy to just merge "feature description, motivation and pitch" into a single box.

albanD reviewed (commented):
Looks really nice! Thanks

NicolasHug commented on .github/ISSUE_TEMPLATE/bug-report.yml:
````diff
@@ -0,0 +1,56 @@
+      A clear and concise description of what the bug is.
+
+      ```python
+      Sample code to reproduce the problem
````
Sure, I'll update. Note that this doesn't matter in practice, this is some text that's visible in the form but that gets deleted as soon as the user starts typing. You can see how it works here:

https://github.com/pytorch/vision/issues/new?assignees=&labels=&template=bug-report.yml

NicolasHug committed:
Update on "Replace issue templates with new issue forms - v2"

This PR introduces the new issue forms that replace issue templates.
(This is exactly the same as https://github.com/pytorch/pytorch/pull/65917 which was reverted due to an issue during the import)

This is similar to what was done in torchvision https://github.com/pytorch/vision/pull/4299 and torchaudio, you can see the end result here: https://github.com/pytorch/vision/issues/new/choose (click e.g. on the [bug report](https://github.com/pytorch/vision/issues/new?assignees=&labels=&template=bug-report.yml))

The main new thing is that we can enforce some of the fields to be filled, especially for bug reports. It's also a much cleaner GUI for users IMHO, and we can provide better examples and instructions.

There is still a "blank" template available.

I removed the "Questions" form: we say we close these issues anyway. I replaced it with a direct link to https://discuss.pytorch.org. Since we still have a "blank" template, I think this  covers all previous use-cases properly.





[ghstack-poisoned]

NicolasHug commented:
@NicolasHug has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32947189).

albanD reviewed (approved):
sgtm!

NicolasHug commented:
Looks like the import went fine this time 😅 thanks all for the reviews!

@vadimkantorov I updated the release template to only include one mandatory field.
